adyenv6core/src/com/adyen/v6/facades/DefaultAdyenCheckoutFacade.java

```diff
@@ -1367,26 +1372,34 @@ private OrderModel retrieveOrder(String orderCode) throws InvalidCartException {
         return orderModel;
     }
 
-    private void restoreCartFromOrder(String orderCode) {
+    private void restoreCartFromOrder(String orderCode) throws CalculationException {
         OrderModel orderModel = getOrderRepository().getOrderModel(orderCode);
         if (orderModel == null) {
             LOGGER.error("Could not restore cart to session, order not found!");
-            getCartService().removeSessionCart();
             return;
         }
 
+        //Create new cart and set in session
         CartModel cartModel = getCartFactory().createCart();
```

Comment:
after placing pending order, a new cart is already being created. Can we reuse same cart instead of creating a new one?

Reply:
That new cart may have already been modified, but we could merge it with the itens from the "old" one (the one from the order).
What do you think?

Reply:
If we have option to merge, I think we should go for merge because user should not loose the items he has added in carts. Let's once again confirm tomorrow in meeting.

Reply:
Fixed